### PR TITLE
fixing typos, adding a new tutorial on deutschs algorithm and general fixing up of notebooks 

### PR DESCRIPTION
**Pull request by zohimchandani**
### Description

1. Adding new section on async functionality 
2. New tutorials on single, multi qubits, teleportation and deutschs algorithm 
3. fixing typos

### CODE REVIEW

**Comment by mmvandieren** [comment]
This PR addresses issue [#897](https://github.com/NVIDIA/cuda-quantum/issues/897)

File: docs/sphinx/examples/python/tutorials/deutschs_algorithm.ipynb
Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/deutschs_algorithm.ipynb:1]: nit: empty cell at the end of file. I also see some equations at the end that are rendering funny on github. They're probably fine in the published docs but you should double check before merging.

File: docs/sphinx/examples/python/tutorials/hybrid_qnns.ipynb
Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/hybrid_qnns.ipynb:1]: I'm fine merging this as-is as to not hold up this PR any longer, but I think a follow-up PR should add further explanation to the code and the entire workflow itself here.

File: docs/sphinx/examples/python/tutorials/multiple_qubits.ipynb
Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/multiple_qubits.ipynb:1]: nit: move the 3 in-line comments to their own line and re-run the formatter

File: docs/sphinx/examples/python/tutorials/quantum_bits.ipynb
Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/quantum_bits.ipynb:1]: nit: move in-line comments to their own line and reformat

File: docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb
Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb:42]: nit: Break this up into a 2 line comment so it doesn't overflow

File: docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb
Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb:1]: nit: There are a few in-line comments that should get their own line, as well as a couple of comments that need a space between the # and the first line, should start with a capital (I think only in code cell 2), and end with a period.

Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/quantum_teleporation.ipynb:1]: Just make sure this equation is rendering fine in the published docs before merging, as it isn't on github

```
$$ \begin{aligned}

CX_{01}H_{0}\ket{00} &= CX_{01}\tfrac{1}{\sqrt{2}}(\ket{0}+\ket{1})\ket{0} \ &= CX_{01}\tfrac{1}{\sqrt{2}}(\ket{00}+\ket{10}) \ &= \tfrac{1}{\sqrt{2}}(\ket{00}+\ket{11}) \ &= \ket{\beta_{00}}

\end{aligned} $$
```

File: docs/sphinx/examples/python/tutorials/vqe.ipynb
Comment (anthony-santana) [inline docs/sphinx/examples/python/tutorials/vqe.ipynb:1]: nit: Selfishly, I would make the cudaq optimizer the focal point here, then use that last code cell to say "We can also use third-party optimizers ..." . I have no strong preference though so it's up to you

**Review by anthony-santana — APPROVED** [review]
Looks great, thanks Zohim! I left some nits throughout but I think it's ready for merging once we've confirmed these are all generating properly in the published docs.

**Comment by schweitzpgi** [comment]
Zohim, can you use the slash command to start the test bots?

**Pull request closed without merging**
